compiler/codegen/OMRAheadOfTimeCompile.hpp

```diff
@@ -88,6 +89,11 @@ class OMR_EXTENSIBLE AheadOfTimeCompile
 
    void traceRelocationOffsets(uint8_t *&cursor, int32_t offsetSize, const uint8_t *endOfCurrentRecord, bool isOrderedPair);
 
+   /**
+    * Do project-specific processing of an AOT relocation just before combining
+    * it into an IteratedExternalRelocation.
+    */
+   static void interceptAOTRelocation(TR::ExternalRelocation *relocation) { }
```

Comment:
Couple of points here:
1) Why is this implemented on AheadOfTimeCompile instead of ExternalRelocation?

2) I think you could have come up with a more informative name here (at least there's a comment to describe it :) ). It's especially weird that an extending project is also expected to call this "interceptAOTRelocation()" back in OMR.

So, maybe ExternalRelocation::finalProjectSpecificHandling() or something like that?

Reply:
`AheadOfTimeCompile` is already `OMR_EXTENSIBLE`. I find that the proliferation of files and identically-named classes required for the extensible class pattern make the code harder to work with, so I'd prefer not to convert `ExternalRelocation` if possible. I'm not sure that `ExternalRelocation` is a very natural place for logic either. It feels to me more like a bag of bits. It doesn't even know what type the data is supposed to be, so everything gets cast to `uint8_t*`.

I agree the name could have been better. I was trying to (partially) address @charliegracie's comment quickly, since this change was holding up a bug fix.

You're right as well that there's no reason why a downstream project should have to call the implementation from OMR, since OMR can always just put code next to the call, instead of inside this function. I called OMR from OpenJ9 because of a more general thought that a child project should probably call the parent's implementation unless it specifically knows otherwise.

So what I'd like to do is the following:
- Keep this method on `AheadOfTimeCompile`,
- Document that the default implementation is a no-op,
- Remove the call to OMR from OpenJ9, and
- Rename to sound more like the hook this is, e.g. `beforeCombiningAOTRelocation()`.

Renaming without breaking the build will take three PRs, but the other changes should be easily folded into those.